### ISSUE TITLE
Add toehold docs for Asyncify

### ISSUE THREAD

**Issue by codefromthecrypt** [WebAssembly/binaryen#4971]
The current source of Asyncify is here, but I think the only docs for it are still in Emscripten. This is a little confusing to say look at another org for docs on something defined here. https://emscripten.org/docs/porting/asyncify.html

**Comment by kripken** [comment]
I agree docs in this repo might make sense. There are docs [in the pass itself](https://github.com/WebAssembly/binaryen/blob/main/src/passes/Asyncify.cpp), but maybe we could have a section in the README that links to that? Also I often point people to this blogpost for doc purposes:

https://kripken.github.io/blog/wasm/2019/07/16/asyncify.html

It could also link to the emscripten docs as an example of high-level integration.

A PR with that sounds good to me.
